.github/workflows/swap-azure-web-app-slots.yml

```diff
@@ -0,0 +1,103 @@
+name: Swap Azure Web App Slots
+
+on:
+  workflow_call:
+    secrets:
+      AZURE_APP_SERVICE_SWAP_SERVICE_PRINCIPAL:
+        required: true
+    inputs:
+      cancel-workflow-on-failure:
+        description: When set to "true", will cancel the current workflow run with all jobs if this workflow fails.
+        required: false
+        type: string
+        default: "true"
+      machine-type:
+        required: false
+        type: string
+        default: ubuntu-latest
+        description: The name of the type of machine to run the workflow under.
+      timeout-minutes:
+        required: false
+        type: number
+        default: 360
+        description: Configuration for the timeout-minutes parameter of the workflow. The 360 is GitHub's default.
+      app-name:
+        required: true
+        type: string
+        description: What you see at the top of the blade on the Azure Portal. Can contain uppercase letters too.
+      source-slot-name:
+        required: false
+        type: string
+        default: Staging
+        description: >
+          The slot name of the web app you want to swap the destination slot (of the same web app) with. What you see at
+          the top of the blade on the Azure Portal, when you open the slot, before the app name in parenthesis.
+      destination-slot-name:
+        required: false
+        type: string
+        default: Production
+        description: >
+          The slot name of the web app you want to swap with the source slot (of the same web app). What you see at the
+          top of the blade on the Azure Portal, when you open the slot, before the app name in parenthesis.
+      resource-group-name:
+        required: true
+        type: string
+        description: Name of the resource group.
+      application-insights-resource-id:
+        required: true
+        type: string
+        description: >
+          ID of the Application Insights resource that the release annotation for the swap should be added to.
+          This can e.g. be looked up on the Azure Portal under the given AI resource's Overview page -> JSON View.
+
+jobs:
+  swap-azure-web-app-slots:
+    runs-on: ${{ inputs.machine-type }}
+    name: Swap Azure Web App Slots
+    defaults:
+      run:
+        shell: pwsh
+    timeout-minutes: ${{ inputs.timeout-minutes }}
+    steps:
+    - name: Login to Azure
+      # v1.4.6
+      uses: azure/login@92a5484dfaf04ca78a94597f4f19fea633851fa2
+      with:
+        creds: ${{ secrets.AZURE_APP_SERVICE_SWAP_SERVICE_PRINCIPAL }}
+        enable-AzPSSession: true
```

Comment:
We use this in multiple places now. Refactor it to a common internal action, like `checkout`.

Reply:
We already have an action called `checkout`. Do you mean, that I should add it to that action as an optional step?

Reply:
Encapsulate this code here into a composite action, similar to how we did with our `checkout` action, and use it instead of `azure/login` everywhere.

Reply:
Oh okay, I misunderstood.